### DISCUSSION TITLE
Route Naming Convention

### DISCUSSION THREAD

**Discussion by manfe** [adonisjs/core#3673] [Help]
Is there any route naming convention for multiple words resources. eg. `UserLogs`

What option should I use:

- Route.resource('user_logs', 'UserLogsController')
- Route.resource('userLogs', 'UserLogsController')
- Route.resource('user-logs', 'UserLogsController')

This is something that I didn't find in the documentation.

But would be nice to have such and example.

**Comment by Julien-R44** [comment]
Hello 👋 

This is not directly related to AdonisJS but more to REST API design and best practices itself, that's why you won't find it in the documentation of AdonisJS.

The most common practice is to have all URLs in lower case and also to use hyphen to separate words.
You'll find a lot of articles and docs about this on the internet, but here is what I think is a good introduction : 
https://stackoverflow.com/questions/10302179/hyphen-underscore-or-camelcase-as-word-delimiter-in-uris

**Reply by manfe** [reply]
Forgot to mention that the question was related to only for API. 

But yeah, for end-user applications the use of `-` (dashes) is the recommended.

`/path-to-whatever/you-want-to`

I am just asking this because Adonis, looks a lot with Rails and Rails uses `_` (underscores)

`/path_to_whatever/you_want_to`
